lib/resolve-config.js

```diff
@@ -44,11 +45,69 @@ module.exports = (
     assets: assets ? castArray(assets) : assets,
     milestones: milestones ? castArray(milestones) : milestones,
     successComment,
-    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
+    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY, NO_PROXY),
   };
 };
 
-function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY) {
+function shouldProxy(gitlabUrl, NO_PROXY) {
```

Comment:
Not strictly required by the license, but still nice to give attribution:

```suggestion
// Copied from Rob Wu's great proxy-from-env library: https://github.com/Rob--W/proxy-from-env/blob/96d01f8fcfdccfb776735751132930bbf79c4a3a/index.js#L62
function shouldProxy(gitlabUrl, NO_PROXY) {

```